leshan-integration-tests/src/test/java/org/eclipse/leshan/integration/tests/IntegrationTestHelper.java

```diff
@@ -246,6 +248,12 @@ public void deregisterClient() {
 
     public void dispose() {
         deregisterClient();
+        if (null != client) {
+            client.destroy(false);
+        }
+        if (null != server) {
+            server.destroy();
+        }
```

Comment:
In all case, you need to change @after : to remove `stop()` or to replace it by `destroy()`.  (I do a call hierarchy, we should change it at 9 locations.)

About using Rule, you can do that in a second step, if you think this can simplify tests.